### PR TITLE
refactor(search): add RepositoryKind::GitConfig to make config based repository explicit

### PR DESCRIPTION
**Pull request by xxs-wallace**
one of https://github.com/TabbyML/tabby/issues/2783

### CODE REVIEW

File: ee/tabby-webserver/src/service/repository/mod.rs [hunk 1]
```diff
@@ -105,15 +105,15 @@ impl RepositoryService for RepositoryServiceImpl {
...
+        if let RepositoryKind::GitConfig = kind {
             if let Ok(index) = config_id_to_index(id) {
                 let config = &self.config[index];
                 return repository_config_to_repository(index, config);
             }
         }
 
         match kind {
-            RepositoryKind::Git => self.git().get_repository(id).await,
+            RepositoryKind::Git | RepositoryKind::GitConfig => self.git().get_repository(id).await,
```
Comment (wsxiaoys) [inline ee/tabby-webserver/src/service/repository/mod.rs:116]: will this be working? can you test this behavior locally?

see https://tabby.tabbyml.com/docs/administration/context/#adding-a-repository-through-configuration-file for a `config.toml` based configuration, and verify it in `Code Browser`

**Review by wsxiaoys — CHANGES_REQUESTED** [review]

**Comment by wsxiaoys** [comment]
Please also write more informative commit message - `add config` is too abstract to represent the intent behind the PR.

File: ee/tabby-webserver/src/service/repository/mod.rs [hunk 2]
```diff
@@ -105,15 +105,15 @@ impl RepositoryService for RepositoryServiceImpl {
     }
 
     async fn resolve_repository(&self, kind: &RepositoryKind, id: &ID) -> Result<Repository> {
-        if let RepositoryKind::Git = kind {
+        if let RepositoryKind::GitConfig = kind {
             if let Ok(index) = config_id_to_index(id) {
```
Comment (wsxiaoys) [inline ee/tabby-webserver/src/service/repository/mod.rs:109]: if it's an GitConfig but not able to run `config_id_to_index` successfully, we shall bail instead?

**Comment by wsxiaoys** [comment]
> Please also provide jam.dev screen recording for easier communication, see https://github.com/TabbyML/tabby/pull/2847 for an example.

Please add a screen recording, otherwise LGTM

**Comment by xxs-wallace** [comment]
https://jam.dev/c/f0165bd9-0378-430a-8c6d-f6f69b114f3c

**Pull request merged**